assets/differentialabundance_report.Rmd

```diff
@@ -0,0 +1,848 @@
+  exploratory_mad_threshold: null
+  exploratory_main_variable: null
+  versions_file: null                                         # e.g 17_software_versions.yml
+  logo: null
+  css: null
+  citations: null
+  filtering_min_samples: 1
+  filtering_min_abundance: 1
+  filtering_min_proportion: NULL
+  filtering_grouping_var: NULL
+  differential_file_suffix: NULL
+  differential_feature_id_column: NULL
+  differential_feature_name_column: NULL
+  differential_fc_column: NULL
+  differential_pval_column: NULL
+  differential_qval_column: NULL
+  differential_min_fold_change: NULL
+  differential_foldchanges_logged: NULL
+  differential_max_qval: NULL
+  deseq2_test: NULL
+  deseq2_fit_type: NULL
+  deseq2_sf_type: NULL
+  deseq2_min_replicates_for_replace: NULL
+  deseq2_use_t: NULL
+  deseq2_lfc_threshold: NULL
+  deseq2_alt_hypothesis: NULL
+  deseq2_independent_filtering: NULL
+  deseq2_p_adjust_method: NULL
+  deseq2_alpha: NULL
+  deseq2_minmu: NULL
+  deseq2_vs_method: NULL
+  deseq2_shrink_lfc: NULL
+  deseq2_cores: NULL
+  deseq2_vs_blind: NULL
+  deseq2_vst_nsub: NULL
+  gsea_run: false
+  gsea_nperm: NULL
+  gsea_permute: NULL
+  gsea_scoring_scheme: NULL    
+  gsea_metric: NULL 
+  gsea_sort: NULL
+  gsea_order: NULL
+  gsea_set_max: NULL
+  gsea_set_min: NULL
+  gsea_norm: NULL
+  gsea_rnd_type: NULL
+  gsea_make_sets: NULL
+  gsea_median: NULL
+  gsea_num: NULL
+  gsea_plot_top_x: NULL
+  gsea_rnd_seed: NULL
+  gsea_save_rnd_lists: NULL
+  gsea_zip_report: NULL
+  gsea_chip_file: NULL 
+  gsea_gene_sets: NULL
+---
+
+<!-- Load libraries -->
+
+```{r, include=FALSE}
+library(knitr)
+library(yaml)
+library(shinyngs)
+library(plotly)
+library(DT)
+```
+
+```{r include = FALSE}
+# Load the datatables js
+datatable(NULL)
+```
+
+```{r, echo=FALSE}
+
+# this function will be available via shinyngs in a release soon but we can use it here for now
+anova_pca_metadata <- function(pca_coords, pcameta, fraction_explained){
+  # Use 10 components or however many fewer is produced by the PCA
+  
+  last_pc <- 10
+  if (ncol(pca_coords) < last_pc) {
+    last_pc <- ncol(pca_coords)
+  }
+  
+  # Remove non-useful variables (those with 1 value, or N values where N is the
+  # number of samples)
+  
+  pcameta <- pcameta[, chooseGroupingVariables(pcameta), drop = FALSE]
+  
+  # Make a blank matrix to hold the p values
+  
+  pvals <-
+    matrix(
+      data = NA,
+      nrow = ncol(pcameta),
+      ncol = last_pc,
+      dimnames = list(
+        colnames(pcameta),
+        paste(
+          paste("PC", 1:last_pc, sep = ""),
+          " (",
+          fraction_explained[1:last_pc],
+          "%)",
+          sep = ""
+        )
+      )
+    )
+  
+  # Fill the matrix with anova p values
+  
+  for (i in 1:ncol(pcameta)) {
+    for (j in 1:last_pc) {
+      fit <- aov(pca_coords[, j] ~ factor(pcameta[, i]))
+      if ("Pr(>F)" %in% names(summary(fit)[[1]])) {
+        pvals[i, j] <- summary(fit)[[1]][["Pr(>F)"]][[1]]
+      }
+    }
+  }
```

Comment:
I usually stick to for loops for small n (as here- we're only talking about 10s of things) since the speed penalties are too small to notice and they're usually more readable for people who don't spend all their time in R. 

But okay, will do ;-)